src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -1,52 +1,186 @@
 using System;
+using System.IO;
 using System.Reflection;
 
 namespace Datadog.Trace.ClrProfiler.Managed.Loader
 {
     /// <summary>
-    /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+    /// This is the only publi class in this assembly.
+    /// The entire assembly is compiled into the native profiler DLL as a resource.
+    /// This happens both, for the profiler part of the Tracer, and for the actual Profiler.
+    /// The native code then uses this class to call arbitrary managed code:
+    /// It uses IL rewriting to inject a call to <c>Startup.Run(..)</c> and passes a list of assemblies.
+    /// Then, this class loads all of the specified assemblies and calls a well known etry point in those assemblies.
```

Comment:
```suggestion
    /// Then, this class loads all of the specified assemblies and calls a well known entry point in those assemblies.
```

Reply:
Updated in _macrogreg/loader-with-modules-cctor-and-managed-loader_-branch.